src/Build/Graph/ProjectGraphNode.cs

```diff
@@ -0,0 +1,33 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections.Generic;
+using Microsoft.Build.Evaluation;
+
+namespace Microsoft.Build.Graph
+{
+    /// <summary>
+    /// Represents the node for a particular project in a project graph.
+    /// </summary>
+    internal class ProjectGraphNode
+    {
+        // No public creation.
+        internal ProjectGraphNode() { }
+
+        /// <summary>
+        /// Gets an unordered collection of graph nodes for projects which this project references.
+        /// </summary>
+        public IReadOnlyCollection<ProjectGraphNode> ReferencedProjects => Array.Empty<ProjectGraphNode>();
```

Comment:
Ok, so.

We spend a lot of time overpronouncing referenc**ed** and referenc**ing** and other similar words. For this formal construction, can we come up with two words that don't sound similar and use them consistently? It could save a lot of confusion in the future.

Reply:
Maybe `ProjectReferences` (parallel to what goes into the project file) and . . . ?

Or `DependsOnProjects` and `BeforeProjects` by (bad) extension of the target-graph operators? Possibly more confusing though since edges in the target graph have only one anchor node.

Reply:
I anticipated naming being difficult which is why this PR is just that by itself :).

In theory we don't *need* a collection referenc**ing** projects since we have the full graph, but I figured it'd just make for easier traversal. If we dropped it though this discussion becomes moot. Personally I think we should keep it though and just figure out a good name somehow.

And to add more options, we could also list these as `Dependencies` instead as a possibly more generic term. Although I like your `ProjectReferences` since it's very clear about how these were constructed. It might box us in in case there some theoretical future way to construct a graph, but I think that would be fairly low probability considering how prevalent project reference items are.

Regarding referenc**ed** vs referenc**ing** (or dependencies vs dependents), I'm completely open to suggestion. I really like your idea of `ProjectReferences` for referenc**ed**, but I'm not sold on what the name for referenc**ing** should be...

QuickBuild uses parent/children (children depend on their parents), but that's at odds with how people tend to visualize these graphs with the "root" at the top and doesn't really align with anything currently in MSBuild terminology-wise.

Reply:
Every time you say parent/child I get confused and have to think for a second so I'd like to vote against that one :)

Unfortunately, I also don't have a great idea for incoming edges.

* ProjectsThatDependOnThis
* ReverseReferences
* BackReferences